core/lib/zksync_core/src/api_server/web3/pubsub.rs

```diff
@@ -181,18 +231,34 @@ impl PubSubNotifier<(typed::Sink<PubSubResult>, PubSubFilter)> {
                 last_block_number = MiniblockNumber(last_log.block_number.unwrap().as_u32());
                 let notify_latency =
                     PUB_SUB_METRICS.notify_subscribers_latency[&SubscriptionType::Logs].start();
+                let subscribers = self.subscribers.read().await;
+                let mut closed_subscriptions = vec![];
 
-                for (sink, filter) in self.current_subscribers().await {
+                for (sink, filter) in subscribers.values() {
                     for log in &new_logs {
                         if filter.matches(log) {
-                            if sink.notify(Ok(PubSubResult::Log(log.clone()))).is_err() {
+                            if sink
+                                .send(
+                                    SubscriptionMessage::from_json(&PubSubResult::Log(log.clone()))
+                                        .expect("PubSubResult always serializable to json;qed"),
+                                )
+                                .await
```

Comment:
Just to understand better: What are the factors determining whether we will block here? AFAIU, it *could* be related to the [`message_buffer_capacity` parameter](https://docs.rs/jsonrpsee-server/latest/jsonrpsee_server/struct.ServerBuilder.html#method.set_message_buffer_capacity) (1,024 by default; IIUC it's applied per connection, rather than per subscription). In general, are we OK with one slow client blocking others? I guess one other option would be to drop slow clients (e.g., by using [`send_timeout`](https://docs.rs/jsonrpsee-server/latest/jsonrpsee_server/struct.SubscriptionSink.html#method.send_timeout) instead of `send`), and/or handling all clients concurrently (although I'm not entirely sure how the latter approach would impact performance).

Reply:
I've added a hardcoded time limit to `send_timeout` here, should we move it to configuration or change to lower amount (currently 1 second)?

Reply:
I prefer the third solution. 
With an additional actor. 
It allows us not to care about slow clients at all.

The capacity shouldn't be an issue.